src/org/daisy/dotify/formatter/impl/page/RowGroupProvider.java

```diff
@@ -275,4 +276,24 @@ private boolean shouldAddGroupForEmptyContent() {
         }
     }
 
+    /**
+     * Make rows invisible based on the block context and the "display-when" property of the block.
+     */
+    private RowImpl makeInvisible(RowImpl row) {
+        if (invisible) {
+            row = new RowImpl.Builder(row).invisible(true).build();
+        }
+        return row;
+    }
+
+    private List<RowImpl> makeInvisible(List<RowImpl> rows) {
+        if (invisible && rows.size() > 0) {
+            List<RowImpl> newRows = new ArrayList<>();
+            for (RowImpl r : rows) {
+                newRows.add(new RowImpl.Builder(r).invisible(true).build());
+            }
+            rows = newRows;
+        }
+        return rows;
```

Comment:
Code may be simplified - the whole body can be replace by
`return rows.stream().map(this::makeInvisible).collect(toList());`

Reply:
Done. I chose to not replace the whole body though, for the sake of efficiency (not always needed to make a new list).